capsules/src/net/udp/driver.rs

```diff
@@ -358,10 +358,10 @@ impl<'a> SyscallDriver for UDPDriver<'a> {
             _ => Err(ErrorCode::NOSUPPORT),
         };
 
-        if let Err(e) = res {
-            Err((slice, e))
-        } else {
-            Ok(slice)
+        match res {
+            Err(e) => Err((slice, e)),
```

Comment:
Why don't you need to do `e.into()` here? Is it implicitly casting from `Error` to `ErrorCode`?

Reply:
In this case `e` is already an `ErrorCode`. The type of `res` is `Result<Result<(), ErrorCode>, ErrorCode>`. The `ErrorCode` in the `Err()` case can be traced as follows:

`self.apps.enter` returns a `Result<Result<(), ErrorCode>, process::Error>`

then,the call to `map_err()` converts that to `Result<Result(), ErrorCode>, ErrorCode>` by only transforming the type of the `Err` case for the outer `Result`